### PR TITLE
Add some jsdoc types to `lib` and `context`

### PR DESCRIPTION
**Pull request by askoufis**
Was finding myself adding jsdoc types here and there across various branches I'm working on, so I figured it would be easier to merge them into a single PR, and add types to the rest of `lib` and `context`. Since `lib` and `context` are used quite extensively throughout the rest of the codebase, we get a lot of inference elsewhere from just annotating these functions/values.

I've enabled some rules from the jsdoc eslint plugin to keep things consistent. We can look into enabling more of them later if we want. Also I added `eslint --fix .` to the `format` script so both formatting and lint warnings/errors will be autofixed now.

**NOTE**: Not every file in `lib` or `context` needed jsdoc comments as existing dependency types meant that the correct types were already inferred.

### CODE REVIEW

File: packages/sku/scripts/start.js [hunk 1]
```diff
@@ -103,57 +103,60 @@ const hot = process.env.SKU_HOT !== 'false';
...
-        const matchingSiteName = getSiteForHost(req.hostname);
-
-        const matchingRoute = routes.find(({ route, siteIndex }) => {
-          if (
-            typeof siteIndex === 'number' &&
-            matchingSiteName !== sites[siteIndex].name
-          ) {
-            return false;
+      middlewares.push(
+        /** @type {import("express").RequestHandler} */
```
Comment (askoufis) [inline packages/sku/scripts/start.js:107]: The github diff looks much better if you hide whitespace.

File: packages/sku/context/index.js [hunk 2]
```diff
@@ -7,9 +7,13 @@ const args = require('../config/args');
 const defaultSkuConfig = require('./defaultSkuConfig');
 const defaultClientEntry = require('./defaultClientEntry');
 const validateConfig = require('./validateConfig');
+
 const defaultCompilePackages = require('./defaultCompilePackages');
 const isCompilePackage = require('../lib/isCompilePackage');
 
+/** @typedef {import("../").SkuConfig} SkuConfig */
+
+/** @returns {{appSkuConfig: SkuConfig, appSkuConfigPath: string | null}} */
```
Comment (mrm007) [inline packages/sku/context/index.js:16]:
```suggestion
/** @returns {{ appSkuConfig: SkuConfig, appSkuConfigPath: string | null }} */
```

File: packages/sku/lib/allocatePort.js [hunk 3]
```diff
@@ -2,7 +2,10 @@ const { yellow, bold } = require('chalk');
 const getPort = require('get-port');
 const debug = require('debug')('sku:allocatePort');
 
-module.exports = async ({ port, host }) => {
+/**
+ * @param {{port?: number, host?: string}}
+ * */
```
Comment (mrm007) [inline packages/sku/lib/allocatePort.js:7]:
```suggestion
 * @param {{ port?: number, host?: string }}
 */
```

File: packages/sku/lib/banner.js [hunk 4]
```diff
@@ -2,7 +2,13 @@ const chalk = require('chalk');
 const wrap = require('wrap-ansi');
 const identString = require('indent-string');
 
-module.exports = (type, heading, messages = []) => {
+/**
+ * @param {'info' | 'error' | 'warning'} type
+ * @param {string} heading
+ * @param {string[]} messages
+ *
+ * */
```
Comment (mrm007) [inline packages/sku/lib/banner.js:10]:
```suggestion
 */
```

There are a few more instances of this, but I'll stop here.

File: packages/sku/lib/runESLint.js [hunk 5]
```diff
@@ -8,6 +8,9 @@ const {
 
 const extensions = [...tsExtensions, ...jsExtensions].map((ext) => `.${ext}`);
 
+/**
+ * @param {{fix?: boolean, paths?: string[]}}
```
Comment (mrm007) [inline packages/sku/lib/runESLint.js:12]:
```suggestion
 * @param {{ fix?: boolean, paths?: string[] }}
```

File: packages/sku/lib/runPrettier.js [hunk 6]
```diff
@@ -12,6 +12,9 @@ const prettierConfigPath = path.join(
   '../config/prettier/prettierConfig.js',
 );
 
+/**
+ * @param {{write?: boolean, listDifferent?: boolean, paths?: string[]}}
```
Comment (mrm007) [inline packages/sku/lib/runPrettier.js:16]:
```suggestion
 * @param {{ write?: boolean, listDifferent?: boolean, paths?: string[] }}
```

**Review by mrm007 — APPROVED** [review]
A bit of formatting pedantry, but otherwise great work 👍

**Comment by askoufis** [comment]
> A bit of formatting pedantry, but otherwise great work 👍

Maybe worth enabling some of the rules in https://www.npmjs.com/package/eslint-plugin-jsdoc?

**Pull request merged**